yeps/yep-310.md

```diff
@@ -19,21 +20,48 @@ Daemons implementing this trait allow measurement to be software triggered over
-This software triggered measurement pattern is very common.
+For example, you might wish to wait for a motor to arrive at its position before telling a sensor to make a measurement.
 This trait standardizes this common sensor interface pattern within the yaq ecosystem.
 
+Note that this trait builds on the is-sensor trait defined in [YEP-302](https://yeps.yaq.fyi/302/).
+That trait defines a standard way of accessing the sensor data and metadata.
+The scope of this trait is only the software triggering feature.
+
 # Proposal
 
 This trait requires is-sensor: see [YEP-302](https://yeps.yaq.fyi/302/).
 
-This trait defines the concept of "looping".
-When a sensor is looping, it should self-trigger the moment the previous acquisition is done.
-In certain cases, setting a sensor to looping makes client implementations much easier.
-Daemons will remain always busy while looping.
-
-This trait defines the concept of a "measurement_id".
-This is an integer that should increase every time the measure message is called.
-`get_measured` returns this measurement_id as a channel.
+The principle feature of the has-measure-trigger is the addition of the `measure` method.
+This method can be called to initiate a measurement.
+Because measurements might take some time to complete, the `measure` method does not wait to return the results of the measurement.
+Instead, it merely returns the `measurement_id` of the measurement that was initiated (the "next" measurement).
+Clients will use the messages defined by [YEP-302](https://yeps.yaq.fyi/302/) to access the data once the measurement is complete.
+Clients will be assured of measurement order by comparing the `measurement_id` returned by `get_measured` to tme `measurement_id` that was returned by `measure` previously.
+
+has-measure-trigger does not change the fundamental nature of `measurement_id` as described by [YEP-302](https://yeps.yaq.fyi/302/): `measurement_id` increments when the channel values are changed.
+That is, when a measurement completes.
+The `measurement_id` returned by `measure` can be thought of as the id of the NEXT measurement.
+The value returned by `measure` will never be the same as returned by `get_measurement_id`.
+
+The fact that a measurement can be initiated implies that daemons implementing has-measure-trigger can, at some times, be not measuring.
+We tie the state of busy to the act of measurement.
+A daemon will remain busy as long as it is actively measuring and will release busy once it becomes idle again.
+This is natural in many orchestration situations where software will want to wait for all sensors to finish measuring before proceeding.
+
+It's very important that the behavior of the `measure` method is consistent, especially with respect to order.
+One crucial detail: what happens when `measure` is called during a measurement?
+In this case daemons implementing has-measure-trigger MUST return the `measurement_id` corresponding to the currently ongoing measurement.
```

Comment:
I believe this has to return the `measurement_id` of the previous measurement not of the current one.  See above.

Reply:
There may have to be some note that it is possible that a call to `measure` , then calling `get_measured`, may not necessarily provide the original measurement.  I.e., it may simply hold off until fresh data arrives and is processed.  Unless we want to codify that it must do so.

Reply:
> There may have to be some note that it is possible that a call to measure , then calling get_measured, may not necessarily provide the original measurement. I.e., it may simply hold off until fresh data arrives and is processed. Unless we want to codify that it must do so.

So sorry @kameyer226, I'm not smart enough to parse this statement. I think my confusion hinges on your definition of "original" and "fresh". What does it mean to "hold off"? Some clarification would be greatly appreciated!

Reply:
The get_measured might simply not return anything at all until the acquisition completes...either that or it times out.

Reply:
OK, that's helpful clarification. Unfortunately I have a further question. Are you referring in particular to the very first measurement? Or are you thinking about a more common situation.

Reply:
I should remember to say THANK YOU for your valuable feedback @kameyer226

Reply:
Thank you for all your hard work on this project @untzag    ⭐

Reply:
regarding line 36:  what happens if `measure` is called multiple times before enough data can be collected for a single measurement ?

Reply:
The chronology table is nice